src/utils/messages.ts

```diff
@@ -30,18 +31,43 @@ export const compareMessagesForGrouping = (
   const sendingStatus = (currMessage as UserMessage)?.sendingStatus || '';
   const isAcceptable = sendingStatus !== 'failed';
   return [
-    isSameGroup(prevMessage, currMessage, currentChannel) && isAcceptable,
-    isSameGroup(currMessage, nextMessage, currentChannel) && isAcceptable,
+    isSameGroup(prevMessage, currMessage, currentUserId) && isAcceptable,
+    isSameGroup(currMessage, nextMessage, currentUserId) && isAcceptable,
   ];
 };
 
 export const getMessageCreatedAt = (message: BaseMessage) => format(message.createdAt, 'p');
 
+// Group current user's messages together. The current user's messages
+// may not have their userId on it, and if not, we assume that messages w/ a
+// local send status is the current user's as well.
+// Given the above is true, group by timestamp
+export const areBothFromMyUserAndInSameGroup = (
+  message: CoreMessageType,
+  comparingMessage: CoreMessageType,
+  currentUserId?: string
+) => {
+  if (!currentUserId || !message.createdAt || !comparingMessage.createdAt) return false;
+
+  const isFirstMessageByMe = getIsByMe(currentUserId, message);
+  const isSecondMessageByMe = getIsByMe(currentUserId, comparingMessage);
+
+  if (isFirstMessageByMe && isSecondMessageByMe) {
+    return getMessageCreatedAt(message) === getMessageCreatedAt(comparingMessage);
+  }
+
+  return false;
+}
+
 export const isSameGroup = (
   message: CoreMessageType,
   comparingMessage: CoreMessageType,
-  currentChannel?: GroupChannel,
+  currentUserId?: string
```

Comment:
What was the group being used for before?

Reply:
It was used previously to determine grouping by `read` status, which we removed cause it was buggy on some clients(jinen's) and we weren't using read status anyways.
https://github.com/sendbird/sendbird-uikit-react/blob/main/src/utils/messages.ts